src/System.Private.CoreLib/shared/System/BitOps.cs

```diff
@@ -355,5 +355,210 @@ int SoftwareFallback(ulong v)
             }
 #endif
         }
+
+        /// <summary>
+        /// Reads whether the specified bit in a mask is set.
+        /// Similar in behavior to the x86 instruction BT.
+        /// </summary>
+        /// <param name="value">The value.</param>
+        /// <param name="bitOffset">The ordinal position of the bit to read.
+        /// Any value outside the range [0..7] is treated as congruent mod 8.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool ExtractBit(byte value, int bitOffset)
+            // For bit-length N, it is conventional to treat N as congruent modulo-N under the shift operation.
+            // So for uint, 1 << 33 == 1 << 1, and likewise 1 << -46 == 1 << +18.
+            // Note -46 % 32 == -14. But -46 & 31 (0011_1111) == +18. 
+            // So we use & not %.
+            => ExtractBit((uint)value, bitOffset & 7);
+
+        /// <summary>
+        /// Reads whether the specified bit in a mask is set.
+        /// Similar in behavior to the x86 instruction BT.
+        /// </summary>
+        /// <param name="value">The value.</param>
+        /// <param name="bitOffset">The ordinal position of the bit to read.
+        /// Any value outside the range [0..31] is treated as congruent mod 32.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool ExtractBit(uint value, int bitOffset)
+            => (value & (1u << bitOffset)) != 0;
+
+        /// <summary>
+        /// Reads whether the specified bit in a mask is set.
+        /// Similar in behavior to the x86 instruction BT.
+        /// </summary>
+        /// <param name="value">The value.</param>
+        /// <param name="bitOffset">The ordinal position of the bit to read.
+        /// Any value outside the range [0..7] is treated as congruent mod 8.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool ExtractBit(int value, int bitOffset)
+            => (value & (1 << bitOffset)) != 0;
+
+        /// <summary>
+        /// Clears the specified bit in a mask and returns whether it was originally set.
+        /// Similar in behavior to the x86 instruction BTR.
+        /// </summary>
+        /// <param name="value">The value.</param>
+        /// <param name="bitOffset">The ordinal position of the bit to clear.
+        /// Any value outside the range [0..7] is treated as congruent mod 8.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool ClearBit(ref byte value, int bitOffset)
+        {
+            uint mask = 1u << (bitOffset & 7);
+
+            bool btr = (value & mask) != 0;
+            value = (byte)(value & ~mask);
+
+            return btr;
+        }
+
```

Comment:
Callsites for `ClearBit`
https://raw.githubusercontent.com/dotnet/roslyn/367e08d8f9af968584d5bab84756eceda1587bd9/src/Workspaces/Core/Portable/Utilities/CompilerUtilities/ImmutableHashMap.cs
https://raw.githubusercontent.com/dotnet/corefx/bd414c68872c4e4c6e8b1a585675a8383b3a9555/src/System.DirectoryServices.AccountManagement/src/System/DirectoryServices/AccountManagement/Utils.cs
        TBD https://raw.githubusercontent.com/dotnet/iot/93f2bd3f2a4d64528ca97a8da09fe0bfe42d648f/src/devices/Mcp23xxx/Mcp23xxx.cs